lib/Mojolicious/Plugin/WebSocketProxy.pm

```diff
@@ -58,14 +58,18 @@ sub register {
         die 'No actions found!';
     }
 
+    # For backwards compatibility, we always want to add a plain JSON::RPC backend
     $dispatcher_config->add_backend(default => Mojo::WebSocketProxy::Backend->new(
+        type => 'jsonrpc',
         url => delete $config->{url},
-    ));
+    )) unless exists $config->{backends}{default};
```

Comment:
This won't exist yet because the code adding them is the block just afterwards. So if the `unless` logic is wanted, it needs to be after the foreach loop currently just below.

Reply:
Probably needed a better comment - the idea is that we'd allow people to provide an explicit `default => {...}` setting, allowing the removal of JSON::RPC if desired, but existing code should keep working.

Since it's using the config value, the data should be available - however! There is a clear bug here which I didn't catch earlier: the `exists` clause will autovivify `$config->{backends}` and mean the next `if()` clause which loops over those config values always runs. Will fix, thanks.

Reply:
Ooooh right - that's the incoming config structure, not the one we're building up. I misread.